### PR TITLE
[WIP] [UI Tests] Verify the page main menu items tests

### PR DESCRIPTION
**Pull request by TejaswiKarasani**
[This](https://testrail.stage.mozaws.net/index.php?/cases/view/233849) is a UI test from Smoke and Sanity Suite of test rail
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt [hunk 1]
```diff
@@ -64,4 +66,192 @@ class SmokeTest {
...
+    @Ignore
+    @Test
+    fun verifyPageMainMenuItemsListLandscapeMode() {
+        val defaultWebPage = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+        activityTestRule.getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE)
+
+        homeScreen {
+            mDevice.waitForIdle(waitingTimeShort)
+            togglePrivateBrowsingModeOnOff()
+            navigationToolbar {
```
Comment (isabelrios) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:202]: I think you could start the test from here and so simplify a little bit more the test

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt [hunk 2]
```diff
@@ -64,4 +66,192 @@ class SmokeTest {
...
+                verifyShortcutNameField(defaultWebPage.title)
+                clickAddShortcutButton()
+                clickAddAutomaticallyButton()
+                verifyShortcutIcon()
+            }.openHomeScreenShortcut(defaultWebPage.title) {
+            }.openThreeDotMenu {
+            }.bookmarkPage {
+                verifySnackBarText("Bookmark saved!")
+            }.openThreeDotMenu {
+                verifyEditBookmarkButton()
```
Comment (isabelrios) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:239]: I don't think this is needed according to the test description...

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt [hunk 3]
```diff
@@ -64,4 +66,192 @@ class SmokeTest {
...
+            }.openEditBookmarks {
+                clickdeleteBookmarkButton()
+            }.confirmBookmarkFolderDeletionAndGoBackToBrowser {
+                verifySnackBarText("Deleted localhost")
```
Comment (isabelrios) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:243]: Same here, all these steps look out of scope for this test and they complicate and add a lot of extra code for now... I would say this is covered by other tests too, @sv-ohorvath do you know? Thanks!

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt [hunk 4]
```diff
@@ -64,4 +66,192 @@ class SmokeTest {
...
+                // verifyShareOverlay()
+            }.closeShareDialogReturnToPage {
+            }.openThreeDotMenu {
+            }.refreshPage {
+                // verifyPageContent(defaultWebPage.content)
+            }.openTabDrawer {
+                closeTabViaXButton(defaultWebPage.title)
+            }.openHomeScreen {
+
+                // PRIVATE BROWSER
```
Comment (isabelrios) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:131]: As we talked on Riot, yes, I think it make sense to split the test for simplicity

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt (hunk 3 above)
Comment (TejaswiKarasani) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:243]: I added this because, I could continue with the below  in private mode as well.

So, added a bookmark and removed it and when that goes with private mode it directly adds a bookmark as the page hasn't been bookmarked already

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt (hunk 4 above)
Comment (TejaswiKarasani) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:131]: So, 4 tests?

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt (hunk 3 above)
Comment (isabelrios) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:243]: If the tests are separated for regular/private mode then there may be a way to have it working?

Comment (TejaswiKarasani) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:243]: Okay

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt (hunk 4 above)
Comment (isabelrios) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:131]: okay, let's try that approach

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt [hunk 5]
```diff
@@ -64,4 +66,186 @@ class SmokeTest {
             }
         }
     }
-}
+
+    @Test
+    fun verifyPageMainMenuItemsListInPortraitNormalMode() {
+        val defaultWebPage = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+
+        homeScreen {
```
Comment (isabelrios) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:74]: You don't need this, you can start with navigationToolbar ...

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt [hunk 6]
```diff
@@ -64,4 +66,186 @@ class SmokeTest {
...
+                // verifyPageContent(defaultWebPage.content)
+            }
+        }
+    }
+
+        @Test
+        fun verifyPageMainMenuItemsListInPortraitPrivateMode() {
+            val defaultWebPage = TestAssetHelper.getGenericAsset(mockWebServer, 1)
+
+            homeScreen {
```
Comment (isabelrios) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:129]: Same as above, not needed

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt [hunk 7]
```diff
@@ -64,4 +66,186 @@ class SmokeTest {
...
+                        clickAddShortcutButton()
+                        clickAddAutomaticallyButton()
+                        verifyShortcutIcon()
+                    }.openHomeScreenShortcut(defaultWebPage.title) {
+                    }.openThreeDotMenu {
+                    }.bookmarkPage {
+                        verifySnackBarText("Bookmark saved!")
+                    }.openThreeDotMenu {
+                    }.sharePage {
+                        // verifyShareAppsLayout()
```
Comment (isabelrios) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:174]: Can't you find here the `Send to Device` text? it would be a way to check that the share layout is shown...

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt [hunk 8]
```diff
@@ -64,4 +66,186 @@ class SmokeTest {
...
+                        // verifyShareOverlay()
+                    }.closeShareDialogReturnToPage {
+                    }.openThreeDotMenu {
+                    }.refreshPage {
+                        // verifyPageContent(defaultWebPage.content)
```
Comment (isabelrios) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:179]: Could you check the text in the url bar after the refresh? thinking about different ways to do the checks and enable these options...

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt (hunk 7 above)
Comment (isabelrios) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:174]: This worked for me in my android device: 
 ```
val sendToDeviceTitle = mDevice.findObject(UiSelector()
                        .instance(0)
                        .className(TextView::class.java))
                sendToDeviceTitle.waitForExists(TestAssetHelper.waitingTime)
```

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt (hunk 8 above)
Comment (TejaswiKarasani) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:179]: Okay

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt (hunk 7 above)
Comment (TejaswiKarasani) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:174]: Ok I will use this

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt (hunk 5 above)
Comment (TejaswiKarasani) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:74]: Okay

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt [hunk 9]
```diff
@@ -64,4 +66,186 @@ class SmokeTest {
...
+            }.closeShareDialogReturnToPage {
+            }.openThreeDotMenu {
+            }.refreshPage {
+                // verifyPageContent(defaultWebPage.content)
+            }
+        }
+    }
+
+        @Test
+        fun verifyPageMainMenuItemsListInPortraitPrivateMode() {
```
Comment (isabelrios) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:126]: Just realized that the test title should end with Test..`verifyPageMainMenuItemsListInPortraitPrivateModeTest`

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/ThreeDotMenuMainRobot.kt [hunk 10]
```diff
@@ -106,6 +109,25 @@ class ThreeDotMenuMainRobot {
     fun verifyShareALinkTitle() = assertShareALinkTitle()
     fun verifyWhatsNewButton() = assertWhatsNewButton()
     fun verifyAddFirefoxHome() = assertAddToFirefoxHome()
+    fun verifyAddToMobileHome() = assertAddToMobileHome()
+    fun verifyDesktopSite() = assertDesktopSite()
+
+    fun verifyThreeDotMainMenuItems() {
```
Comment (isabelrios) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/ThreeDotMenuMainRobot.kt:115]: This method will fail in landscape...not all options are visible, you should split it in, at least two, before swiping you see and can verify: 
Settings
Finde in Page
Add to top sites
Add to Home screen
Save to collection
Destop site
And the start, share, reload buttons

Then if you scroll down you see:
Add-ons
History
Bookmarks
Synced tabs

If you try to check all of them at the same time, the test fails

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt (hunk 9 above)
Comment (TejaswiKarasani) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:126]: okay

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/ThreeDotMenuMainRobot.kt (hunk 10 above)
Comment (TejaswiKarasani) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/ThreeDotMenuMainRobot.kt:115]: okay

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt (hunk 4 above)
Comment (TejaswiKarasani) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt]: @isabelrios I will open other PR, this one got messed up

Comment (isabelrios) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt]: yes @TejaswiKarasani, sounds good. Closing this one then. Thanks

**Comment by isabelrios** [comment]
Closing as per comments above, a new one will be open
